src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/BookBinder.java

```diff
@@ -57,6 +57,20 @@ protected MachineRecipe findNextRecipe(BlockMenu menu) {
 
                 // Just return if no enchantments exist. This shouldn't ever happen. :NotLikeThis:
                 if (enchantments.size() > 0) {
+                    // If any of the books contain an enchantment level higher that the configured maximum: don't consume the items
+                    for (Map.Entry<Enchantment, Integer> entry : storedItemEnchantments.entrySet()) {
+                        int maxLevel = bypassVanillaMaxLevel.getValue() ? customMaxLevel.getValue() : entry.getKey().getMaxLevel();
+                        if (entry.getValue() > maxLevel) {
+                            return null;
+                        }
+                    }
+                    for (Map.Entry<Enchantment, Integer> entry : storedTargetEnchantments.entrySet()) {
+                        int maxLevel = bypassVanillaMaxLevel.getValue() ? customMaxLevel.getValue() : entry.getKey().getMaxLevel();
+                        if (entry.getValue() > maxLevel) {
+                            return null;
+                        }
+                    }
+
```

Comment:
not a big fan of that we have 2 for loops with the same exact code in it

Reply:
Agreed, it has a helper function now. Is it better like this?